packages/web3-providers/src/types/RSS3.ts

```diff
@@ -58,41 +68,240 @@ export namespace RSS3BaseAPI {
+        withdraw: TransactionMetadata
+        swap: SwapMetadata
+        trade: TradeMetadata
+        liquidity: LiquidityMetadata
+        poap: PoapMetadata
+        post: PostMetadata
+        revise: PostMetadata
+        // TODO Not official documented
+        comment: PostMetadata
+        // TODO Not official documented
+        share: PostMetadata
+        profile: ProfileMetadata
+        // TODO Not official documented
+        follow: null
+        // TODO Not official documented
+        unfollow: null
+        // TODO Not official documented
+        like: null
+        propose: ProposeMetadata
+        vote: VoteMetadata
+        launch: LaunchMetadata
+        donate: DonateMetadata
+    }
+
+    type TypeMap = {
+        [Tag.Transaction]: Type.Transfer | Type.Mint | Type.Burn
+        [Tag.Exchange]: Type.Withdraw | Type.Deposit | Type.Swap | Type.Liquidity
+        [Tag.Collectible]: Type.Transfer | Type.Trade | Type.Mint | Type.Burn | Type.Poap
+        [Tag.Social]: Type.Post | Type.Revise | Type.Comment | Type.Share | Type.Profile
+        [Tag.Donation]: Type.Launch | Type.Donate
+        [Tag.Governance]: Type.Propose | Type.Vote
+    }
+
+    export interface ActionGeneric<T extends Tag, P extends TypeMap[T] = TypeMap[T]> {
+        tag: T
+        type: P
+        index: number
         address_from?: string
         address_to?: string
-        metadata?: Metadata
+        metadata?: MetadataMap[P]
         platform?: PLATFORM
         related_urls?: string[]
     }
 
+    export interface AddressStatus {
+        update_at: RFC3339Datetime
+        address: string
+    }
+
     export type NETWORK =
         | 'ethereum'
-        | 'ethereum_classic'
+        // Discard
+        // | 'ethereum_classic'
         | 'binance_smart_chain'
         | 'polygon'
         | 'zksync'
         | 'xdai'
-        | 'arweave'
+        // Ignore this, since it's now supported runtime network
+        // | 'arweave'
         | 'arbitrum'
         | 'optimism'
         | 'fantom'
         | 'avalanche'
         | 'crossbell'
 
+    export const MaskNetworkMap: Record<NETWORK, ChainId> = {
+        ethereum: ChainId.Mainnet,
+        binance_smart_chain: ChainId.BSC,
+        polygon: ChainId.Matic,
+        zksync: ChainId.ZKSync_Alpha_Testnet,
+        xdai: ChainId.xDai,
+        arbitrum: ChainId.Arbitrum,
+        optimism: ChainId.Optimism,
+        fantom: ChainId.Fantom,
+        avalanche: ChainId.Avalanche,
+        crossbell: ChainId.Crossbell,
+    }
+
     export type PLATFORM = 'mirror' | 'lens' | 'gitcoin' | 'snapshot' | 'uniswap' | 'binance' | 'crossbell'
 
-    export type TAG = 'social' | 'transaction' | 'exchange' | 'collectible' | 'donation' | 'governance'
+    export enum Tag {
+        Collectible = 'collectible',
+        Donation = 'donation',
+        Exchange = 'exchange',
+        Governance = 'governance',
+        Social = 'social',
+        Transaction = 'transaction',
+    }
 
     export type TYPE =
```

Comment:
The following one replaces it, isn't it?

Reply:
Indeed, it's unused.